tests/ztest/src/ztest.c

```diff
@@ -198,7 +198,7 @@ static int run_test(struct unit_test *test)
 
 	TC_START(test->name);
 	k_thread_create(&ztest_thread, thread_stack,
-			K_THREAD_STACK_SIZEOF(thread_stack),
```

Comment:
Something is really wrong here.
thread_stack is defined as: 

```
K_THREAD_STACK_DEFINE(thread_stack, CONFIG_ZTEST_STACKSIZE +
 			     CONFIG_TEST_EXTRA_STACKSIZE);
```

K_THREAD_STACK_SIZEOF(thread_stack), by definition, returns the same value passed to K_THREAD_STACK_DEFINE. It should already be  CONFIG_ZTEST_STACKSIZE + CONFIG_TEST_EXTRA_STACKSIZE.
Why are you getting a different result?